### PR TITLE
[DOC] provide proper mathematical description in docstring of sliding window splitter.

### LINKED ISSUE #7196: [ENH] provide proper mathematical description in docstring of splitters
Many of the splitters are poorly described, in terms of formalism, and in terms of parameters. We should write proper formal descriptions of the splitters.

For a complete description, the docstring should describe:

* what exactly the training folds are
* what exactly the test folds are
* how many splits there are

Descriptions of the folds should be unambiguous in case of irregular indices (if applicable), i.e., are we getting all indices within a specific interval, or are we getting specific indices?

Further, all parameters should be precisely explained.

An example for this is this PR for `SlidingWindowSplitter`:
https://github.com/sktime/sktime/pull/7195

Splitters with incomplete description:

- [ ] `ExpandingCutoffSplitter`
- [ ] `ExpandingGreedySplitter`
- [ ] ExpandingWindowSplitter`
- [ ] `SameLocSplitter`
- [ ] `SingleWindowSplitter`
- [ ] `TemporalTrainTestSplitter`, and `temporal_train_test_split`

### PR DESCRIPTION
**Pull request by keitaVigano**
#### Reference Issues/PRs

Towards #7196 

#### What does this implement/fix? Explain your changes.

I wrote a better documentation for this splitter.

#### Does your contribution introduce a new dependency? If yes, which one?

No.

#### What should a reviewer concentrate their feedback on?

Making sure that the documentation is written correctly.

#### Did you add any tests for the change?

No.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [ ] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
This is great and already a significant improvement!

Only small requests for improvement:

* the docstring should say that there is only a single pair of training/test window, because splitters could have multiple.
* the docstring should say which indices exactly are in the training set. Currently only a "starting point" and "end point" are mentioned.
* what happens if `fh` contains negative values? That would result in an invalid split, would it not?
* in the example, I would also simply list the numbers or indices corresponding to the picture, for maximum clarity.

File: sktime/split/singlewindow.py [hunk 1]
```diff
@@ -34,22 +34,49 @@
...
+    The training set is defined based on a "window".
+    The endpoint of the training set is determined by
+    `the size of the time series - fh[-1] - 1`,
+    and the starting point is calculated as `endpoint - window_length + 1`.
+    If the starting point is negative, it will be set to 0.
+
+    If the time points in the data are :math:`(t_1, t_2, \ldots, t_N)`, the training
+    windows will be all indices in the intervals
+
+    .. math:: [t_{N-fh[-1] - w}, \ldots, t_{N-fh[-1] - 1}]
```
Comment (fkiraly) [inline sktime/split/singlewindow.py:48]: is this correct? That is, is the splitter `iloc` rather than `loc` based? Good to know!

**Review by fkiraly — CHANGES_REQUESTED** [review]
Great, this has added the required level of detail!

However, something cannot be correct here, since the `window_length` claims it can be `timedelta`. It is not clearly stated what happens when `window_length` is `timedelta`. That would be inconsistent with the `int` case as well.

Could you kindly double check that the window applies to `iloc`, if the `window_length` is `int`, and whether it is `loc` or what else happens if it is `timedelta`?

**Comment by fkiraly** [comment]
ready for a review?

File: sktime/split/singlewindow.py [hunk 2]
```diff
@@ -34,22 +34,58 @@
...
-    relative to the end of the training window.
-    It will contain as many indices
+    The training set is defined based on a "window".
+    The endpoint of the training set is determined by
+    `the size of the time series - fh[-1] - 1`,
+    and the starting point is calculated as `endpoint - window_length + 1`.
+    If the starting point is negative, it will be set to 0.
+
+    If the time points in the data are :math:`(t_0, t_1, t_2, \ldots, t_{N - 1})`,
+    the training windows will be all indices in the intervals
```
Comment (fkiraly) [inline sktime/split/singlewindow.py:46]: *the inverval*, no? a single interval?

File: sktime/split/singlewindow.py [hunk 3]
```diff
@@ -34,22 +34,58 @@
...
-    Split time series once into a training and test set.
-    See more details on what to expect from this splitter in :class:`BaseSplitter`.
+    Split time series into a single training and test set window.
 
-    Test window is defined by forecasting horizons
-    relative to the end of the training window.
-    It will contain as many indices
+    The training set is defined based on a "window".
+    The endpoint of the training set is determined by
+    `the size of the time series - fh[-1] - 1`,
```
Comment (fkiraly) [inline sktime/split/singlewindow.py:41]: length, not size, right?

File: sktime/split/singlewindow.py (hunk 1 above)
Comment (fkiraly) [reply sktime/split/singlewindow.py:48]: can you confirm that fh is in the index and not subtracxted from $t_N$ instead?

File: sktime/split/singlewindow.py [hunk 4]
```diff
@@ -34,22 +34,58 @@
...
     as there are forecasting horizons provided to the ``fh`` argument.
-    For a forecasating horizon :math:`(h_1,\ldots,h_H)`, the training window will
-    consist of the indices :math:`(k_n+h_1,\ldots,k_n+h_H)`.
+    In particularly, they will be equal to the endpoint plus the forecasting horizon.
+
+    For a forecasting horizon :math:`(h_1,\ldots,h_H)`, the test indices
+    will consist of the indices :math:`(k+h_1,\ldots,k+h_H)`,
+    where k is the end of the training window.
+
+    **Important Notes:**
```
Comment (fkiraly) [inline sktime/split/singlewindow.py:60]: leave a newline, or the bullet point list will not display. Check the rendering by looking at "details" near the readthedocs CI element.

File: sktime/split/singlewindow.py [hunk 5]
```diff
@@ -34,22 +34,58 @@
...
+    - `SingleWindowSplitter` uses positional indexing (`iloc`) for the training and
```
Comment (fkiraly) [inline sktime/split/singlewindow.py:61]: for in-line code, use double backticks

**Review by fkiraly — CHANGES_REQUESTED** [review]
Nice! Left some comments.

**Comment by keitaVigano** [comment]
@fkiraly Could you please review this?

**Review by fkiraly — APPROVED** [review]
Brilliant! Thanks!

Sorry for the delay.

**Pull request merged**